src/Files.App/UserControls/SideBar/SideBarItem.cs

```diff
@@ -20,13 +20,14 @@ public sealed partial class SidebarItem : Control
 		public bool HasChildren => Item?.Children is IList enumerable && enumerable.Count > 0;
 		public bool IsGroupHeader => Item?.Children is not null;
 		public bool CollapseEnabled => DisplayMode != SidebarDisplayMode.Compact;
+		public bool IsFooterItem => Item?.Section == SectionType.Footer;
 
 		private bool hasChildSelection => selectedChildItem != null;
 		private bool isPointerOver = false;
 		private bool isClicking = false;
 		private object? selectedChildItem = null;
 		private ItemsRepeater? childrenRepeater;
-		private ISidebarItemModel? lastSubscriber;
+		private INavigationControlItem? lastSubscriber;
```

Comment:
This seems like quite a strong mixing of domains. INavigationControlItem contains a lot of stuff that the SideBarItem will never need (e.g. SectionType) and I would be quite concerned if we use SideBar for settings and the entries there having such properties. 

Maybe I'm missing something here but ISidebarItemModel was there to decouple it from the other domains since it was designed to be reusable and not contain much "business logic" of file system related stuff.

Reply:
INavigationControlItem previously inherited from ISidebarItemModel.
And ISidebarViewModel was never used else where.
This means separation is useless.

Reply:
So do we revert this once we use Sidebar in other areas? I would argue, only because right now it is not used otherwise doesn't mean that mixing stuff is fine. After all SideBar could have been built as part of the Mainwindow but was not (similar things for the SidebarViewModel).

Reply:
Yeah this is not needed imo. And SidebarItem.Item reference too.
Generalizing would solve this issue.